packages/flutter/test/material/refresh_indicator_test.dart

```diff
@@ -1178,29 +1169,78 @@ void main() {
         ),
       );
 
-      RefreshProgressIndicator getIndicator() {
-        return tester.widget<RefreshProgressIndicator>(
-          find.byType(RefreshProgressIndicator),
-        );
-      }
-
       // Correspond to [_kDragContainerExtentPercentage].
       final double maxPosition = tester.view.physicalSize.height / tester.view.devicePixelRatio * 0.25;
-      for (final double position in positions) {
+      for (final double position in <double>[50.0, 100.0, 150.0]) {
         await tester.fling(find.text('X'), Offset(0.0, position), 1.0);
         await tester.pump();
         positionController.value = position / maxPosition;
         expect(
-          getIndicator().valueColor!.value!.alpha,
+          tester.widget<RefreshProgressIndicator>(find.byType(RefreshProgressIndicator)).valueColor!.value!.alpha,
           valueColorAnimation.value!.alpha,
         );
         // Wait until the fling finishes before starting the next fling.
         await tester.pumpAndSettle();
       }
     }
+  });
+
+  testWidgets('RefreshIndicator default elevation value correctly', (WidgetTester tester) async {
+    final AnimationController positionController = AnimationController(vsync: const TestVSync());
+    addTearDown(positionController.dispose);
+
+    await tester.pumpWidget(
+      MaterialApp(
+        home: RefreshIndicator(
+          onRefresh: refresh,
+          child: ListView(
+            physics: const AlwaysScrollableScrollPhysics(),
+            children: const <Widget>[Text('X')],
+          ),
+        ),
+      ),
+    );
+
+    // Correspond to [_kDragContainerExtentPercentage].
+    final double maxPosition = tester.view.physicalSize.height / tester.view.devicePixelRatio * 0.25;
+    for (final double position in <double>[50.0, 100.0, 150.0]) {
+      await tester.fling(find.text('X'), Offset(0.0, position), 1.0);
+      await tester.pump();
+      positionController.value = position / maxPosition;
+      expect(tester.widget<RefreshProgressIndicator>(find.byType(RefreshProgressIndicator)).elevation, 2.0);
+      // Wait until the fling finishes before starting the next fling.
+      await tester.pumpAndSettle();
+    }
+  });
 
-    for (final Color color in colors) {
-      await testColor(color);
+  testWidgets('RefreshIndicator custom elevation values correctly', (WidgetTester tester) async {
+    for (final double elevation in <double>[0, 0.1, 1, 1.1, 1.59]) {
```

Comment:
Maybe just try `0.0`, `2.0`, and one other positive value? Or is there a reason to test all of these?

Reply:
Just for variety i added these many. But i think 0.0 and 2.0 should be enough.

@justinmc - Also, I did not write a test-case for negative elevation values. Let me add these assertion and a unit test to fortify those use-cases.

cc: @victorsanni